<?xml version="1.0"?>
<robot name="materials">

  <material name="blue">
    <color rgba="0 0 0.8 1"/>
  </material>

  <material name="white">
    <color rgba="1 1 1 1"/>
  </material>


  <link name="base_link">
    <visual>
      <geometry>
        <box size="0.2 0.3 0.6"/>
      </geometry>
      <material name="blue"/>
    </visual>
  </link>

  <link name="right_leg">
    <visual>
      <geometry>
        <box size="0.01 0.01 0.9"/>
      </geometry>
      <origin xyz="0.0 0 0.0" rpy="0.0 0.0 0.0"/>
      
      <material name="white"/>
    </visual>
  </link>
  <joint name="joint1" type="continuous"> 
    <parent link="base_link"/>
    <child link="right_leg"/>
    <axis xyz="0.0 1 0.0"/>
    <origin xyz="0.0 -0.155 0.0" rpy="0.0 0.0 0.0"/>
  </joint>

  
  <link name="top_fan">
    <visual>
      <geometry>
        <box size="0.1 0.01 0.01"/>
      </geometry>
      <origin xyz="0.0 0 0.0" rpy="0.0 0.0 0.0"/>
      
      <material name="white"/>
    </visual>
  </link>

  <joint name="joint2" type="continuous"> 
    <parent link="right_leg"/>
    <child link="top_fan"/>
    <axis xyz="0 0 1"/>
    <origin xyz="0.0 0 0.455" rpy="0.0 0.0 0.0"/>
  </joint>


  <link name="left_shoulder">
    <visual>
      <geometry>
        <box size="0.01 0.01 0.2"/>
      </geometry>
      <origin xyz="0 0.005 0.1" rpy="0.0 0.0 0.0"/>
      
      <material name="white"/>
    </visual>
  </link>

  <joint name="left_shoulder_joint" type="revolute"> 
    <parent link="base_link"/>
    <child link="left_shoulder"/>
    <limit effort="30" velocity="1.0" lower="-1.570795" upper="1.570795" />
    <axis xyz="0 1 0"/>
    <origin xyz="0.0 0.15 0.2" rpy="0.0 0.0 0.0"/>
  </joint>


  <link name="left_wrist">
    <visual>
      <geometry>
        <box size="0.01 0.01 0.2"/>
      </geometry>
      <origin xyz="0 0.005 0.1" rpy="0.0 0.0 0.0"/>
      
      <material name="white"/>
    </visual>
  </link>

  <joint name="left_wrist_joint" type="revolute"> 
    <parent link="left_shoulder"/>
    <child link="left_wrist"/>
    <limit effort="30" velocity="1.0" lower="-1.570795" upper="1.570795" />
    <axis xyz="0 1 0"/>
    <origin xyz="0.0 0. 0.2" rpy="0.0 0.0 0.0"/>
  </joint>


  
</robot>